java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Order/OrdersRepository.java

```diff
@@ -0,0 +1,178 @@
+// Get the List of appetizers:
+    // we are getting back the a List of appetizers ids, we need to get THE appetizer
+    //we use the method from the AppetizerRepository that we are getting the
+    // appetizer by their ids. We get the appetizer and put them on a new list
+    // For Example:
+    // List<Appetizer> appetizerOrders = new ArrayList<>();
+    // Appetizer appetizers = appetizerRepository.gettingAppetizerById(appetizerId);
+    // appetizerOrders.add(appetizers);
+// Get the List of entrees:
+    // we are getting back the List of entrees ids, we need to get THE entree
+    //we use the method from the EntreeRepository that we are getting the
+    // entree by their id. We get the entree and put them on a new list
+    // For Example:
+    // List<Entree> entreeOrders = new ArrayList<>();
+    // Entree entree = entreeRepository.gettingEntreeById(entreeId);
+    // entreeOrders.add(entree);
+// create a Entity:
+    // we need to return a list of appetizers & entrees(The name,cost etc)
+    //then we need to create a entity DTO(Data Transfer Object) to get the necessary fields
+    // we need to return to the customer(the id of the order, customerId and the list of
+    // appetizers & entrees(NOT THERE IDs) that the customer ordered).
+    // For Example:
+    //OrderDto orderDto = new OrderDto(order.getId(), ordersRequestBody.getCustomerId(),
+    // appetizerOrders, entreeOrders);
+//
+    public OrderDto CreateNewOrder(NewOrderRequestBody ordersRequestBody) {
+        // first when the order is created (Inserted into order) we need to find the appetizers
+        String sql = "INSERT INTO \"order\" (customer_id) VALUES (?) RETURNING *";
+        Orders order = jdbcTemplate.queryForObject(sql, new BeanPropertyRowMapper<>(Orders.class),
+                ordersRequestBody.getCustomerId());
+        log.info("Order is Created");
+        // the order is inserted now we take the appetizerId from the requestBody
+        // now we need to find the id by using the method in AppetizerRepository
+        log.info("Looking for the appetizers...");
+        List<Appetizer> appetizerOrders = new ArrayList<>();
+        List<Integer> listOfAppetizerIds = ordersRequestBody.getAppetizerIds();
+        System.out.println("List of appetizersIds: " + listOfAppetizerIds);
+        listOfAppetizerIds.forEach(appetizerId -> {
+            Appetizer appetizers =
+                    appetizerRepository.gettingAppetizerById(appetizerId);
+            System.out.println("Found appetizer " + appetizers);
+            String sqlAppetizer = " INSERT INTO appetizer_ordered(order_id ,appetizer_id) VALUES (?,?) RETURNING *";
+            AppetizerOrdered appetizerOrdered = jdbcTemplate.queryForObject(sqlAppetizer,
+                    new BeanPropertyRowMapper<>(AppetizerOrdered.class),
+                    order.getId(), appetizers.getId());
+            log.info("Appetizer receipt: " + appetizerOrdered);
```

Comment:
it doesn't seem like you are using this `appetizerOrdered` at all. Is this even needed?

Reply:
I was using appetizerOrdered as a receipt. i get the order id and the appetizerId and inserted in appetizerOrdered Table in my database.
So that i can maybe use it for when i am looking for the order and which appetizers the order has.i thought it was a good try. But I can remove it